src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -976,8 +976,9 @@ public void Sort(IComparer<T> comparer)
             }
             Contract.EndContractBlock();
 
-            if( _size > 0) {
+            if (_size > 0) {
                 ArraySortHelper<T>.Sort(_items, 0, _size, comparison);
+                _version++;
```

Comment:
>It makes no sense to indicate that the collections was changed if it really wasn't.

This is the pattern we've done several other places where the version updating is meant more to mean "a modifying method was called" rather than "the collection changed". We should remain consistent with that and make this Sort method have the same versioning behavior as the other existing sort methods.